src/components/ChartModal/BiAxialChartConfig/index.tsx

```diff
@@ -0,0 +1,412 @@
+import {
+    randomString,
+    isFalsyString,
+    unique,
+    isDefined,
+    isTruthyString,
+    getRandomFromList,
+    _cs,
+} from '@togglecorp/fujs';
+
+import SelectInput from '#components/SelectInput';
+import Button from '#components/Button';
+import TextInput from '#components/TextInput';
+import ColorInput from '#components/ColorInput';
+import NumberInput from '#components/NumberInput';
+import { tableauColors } from '#utils/constants';
+import SegmentInput from '#components/SegmentInput';
+
+import { BiAxialChartSettings, NumericOption } from '#types';
+import styles from './styles.css';
+
+type BiAxialChartDataType = 'line' | 'bar';
+
+interface DataTypeOption {
+    key: BiAxialChartDataType;
+    title: string;
+}
+
+const dataTypeOptions: DataTypeOption[] = [
+    {
+        key: 'line',
+        title: 'Line',
+    },
+    {
+        key: 'bar',
+        title: 'Bar',
+    },
+];
+
+const dataTypeKeySelector = (item: DataTypeOption) => item.key;
+const dataTypeLabelSelector = (item: DataTypeOption) => item.title;
+
+type OrderOptionKey = 'asc' | 'dsc';
+
+interface OrderOption {
+    key: OrderOptionKey;
+    title: string;
+}
+
+const orderOptions: OrderOption[] = [
+    { key: 'asc', title: 'ascending' },
+    { key: 'dsc', title: 'descending' },
+];
+
+const orderKeySelector = (item: OrderOption) => item.key;
+const orderLabelSelector = (item: OrderOption) => item.title;
+
+interface BiAxialData {
+    id: string;
+    optionName?: string;
+    color: string;
+    type: BiAxialChartDataType;
+}
+
+interface BiAxialChartItemProps<T> {
+    value: BiAxialData;
+    onValueChange: (val: (values: BiAxialData[]) => BiAxialData[]) => void;
+    index: number;
+    options: NumericOption<T>[];
+    type: BiAxialChartDataType;
+    onToggleChartType: () => void;
+}
+
+function BiAxialChartItem<T>(props: BiAxialChartItemProps<T>) {
+    const {
+        index,
+        value,
+        onValueChange,
+        options,
+        onToggleChartType,
+    } = props;
+
+    // FIXME: memoize
+    const keySelector = (item: NumericOption<T>) => item.key;
+    const labelSelector = (item: NumericOption<T>) => item.title;
+    const groupSelector = (item: NumericOption<T>) => item.category;
+
+    const handleOptionNameChange = useCallback(
+        (optionName: string | undefined) => {
+            onValueChange((values) => {
+                const newValues = [...values];
+                newValues[index] = {
+                    ...newValues[index],
+                    optionName,
+                };
+                return newValues;
+            });
+        },
+        [onValueChange, index],
+    );
+
+    const handleColorChange = useCallback(
+        (color: string) => {
+            onValueChange((values) => {
+                const newValues = [...values];
+                newValues[index] = {
+                    ...newValues[index],
+                    color,
+                };
+                return newValues;
```

Comment:
Lets use the array edit hook we created:
https://github.com/the-deep/client/blob/develop/src/hooks/stateManagement.ts#L31